Storage/api_base.h

```diff
@@ -97,16 +97,20 @@ struct RESTfulRegisterTopLevelInput : RESTfulGenericInput<STORAGE> {
         up_status(up_status) {}
 };
 
-template <typename STORAGE, typename FIELD>
+template <typename STORAGE, typename FIELD, bool READ_ONLY_TRANSACTION>
 struct RESTfulGETInput : RESTfulGenericInput<STORAGE> {
   using mutable_fields_t = MutableFields<STORAGE>;
-  mutable_fields_t fields;
+  using immutable_fields_t = ImmutableFields<STORAGE>;
+  using fields_type_t =
+      typename std::conditional<READ_ONLY_TRANSACTION, immutable_fields_t, mutable_fields_t>::type;
+  constexpr static bool read_only_transaction = READ_ONLY_TRANSACTION;
```

Comment:
When used in `enable_if_t`, it's not immediately clear to me whether this `read_only_transaction` is a bool. Perhaps go the old way of making it a constexpr function (`IsReadOnly() { return true; }`)? The alternative is the syntax that involves `::value` in templated `enable_if_t<*read_only::value>` use cases.